apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -81,8 +94,43 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     assertCanCreateQuestion(res.locals);
+    if (req.query?.qid) {
+      const qidFull = `__drafts__/${req.query?.qid}`;
+      const prompts = await queryRows(
+        sql.select_ai_question_generation_prompts,
+        { qid: qidFull, course_id: res.locals.course.id.toString() },
+        AiGenerationPromptSchema,
+      );
 
-    res.send(AiGeneratePage({ resLocals: res.locals }));
+      if (prompts && prompts.length > 0) {
+        res.locals.question = await queryRow(
+          sql.select_question_by_qid_and_course,
+          { qid: qidFull, course_id: res.locals.course.id },
+          QuestionSchema,
+        );
+        const variant_id = req.query.variant_id ? IdSchema.parse(req.query.variant_id) : null;
```

Comment:
unfortunately, the renderer by default gets from '/' (while maintaining query parameters), so we have to do the parsing hacks here